feat: add faucet page and playground (#167)
onmax/nimiq-developer-center: web-client/faucet.md

---
outline: deep
description: Get free NIM for development and testing on Nimiq Testnet.
---

<script setup>
import FaucetPlayground from '../.vitepress/theme/components/Faucet/FaucetPlayground.vue'
</script>

# Faucet

> The Nimiq Faucet dispenses free NIM for development and testing.

Reviewer: ```suggestion
The Nimiq Faucet provides **testnet NIM only** and is intended for development and testing purposes.
```

Accessible via a web interface or a simple API, it ensures you have the resources to deploy contracts and simulate real-world scenarios cost-free.

## Interactive Playground

> Use the interface below to instantly request funds to your address.

Reviewer: ```suggestion
Paste a **testnet** Nimiq address in the interface below to request funds.
```

<FaucetPlayground />

## Network Details

The faucet currently supports the Nimiq Testnet, offering unlimited requests to facilitate continuous integration and development workflows.

| Feature | Testnet |
| :--- | :--- |
| **URL** | `faucet.pos.nimiq-testnet.com` |
| **Max Amount** | 10,000 NIM |
| **Rate Limit** | Unlimited |
| **Intended Use** | Development, CI/CD, Stress Testing |

## API Reference

> Developers can interact with the faucet programmatically via HTTP requests. This is particularly useful for automated testing pipelines.

Reviewer: isn't HTTPS?

**Endpoint:** `POST https://faucet.pos.nimiq-testnet.com/tapit`

### Parameters

The endpoint accepts a `application/x-www-form-urlencoded` body with the following parameters:

| Parameter | Type | Required | Description |
| :--- | :--- | :--- | :--- |
| `address` | `string` | **Yes** | The recipient's Nimiq address in User-Friendly format (e.g., `NQ...`). |

> | `amount` | `number` | No | The amount of NIM to request. Defaults to `10000`. <br>Max: `110000`. |

Reviewer: isn't the max  10'000?
